src/sonic-eventd/rsyslog_plugin/rsyslog_plugin.h

```diff
@@ -0,0 +1,34 @@
+#ifndef RSYSLOG_PLUGIN_H
+#define RSYSLOG_PLUGIN_H
+
+#include <string>
+#include "syslog_parser.h"
+#include "common/events.h"
+#include "common/logger.h"
+
+using namespace std;
+using namespace swss;
+
+/**
+ * Rsyslog Plugin will utilize an instance of a syslog parser to read syslog messages from rsyslog.d and will continuously read from stdin
+ * A plugin instance is created for each container/host.
+ *
+ */
+
+class RsyslogPlugin {
+public:
+    bool onInit();
+    bool onMessage(string msg);
+    void run();
+    RsyslogPlugin(string module_name, string regex_path);
+private:
+    SyslogParser* m_parser;
```

Comment:
I line with Qi's comments, you may make this shared-ptr too, so it gets deleted when this object gets deleted.

Reply:
Using smart pointer now